Fix broken link in Host Application email

gustavlrsn opened the pull request:
Broken link reported on Slack.

Depends on https://github.com/opencollective/opencollective-frontend/pull/9402

Updates the link in `collective.apply.for.host` email template.

Betree commented on templates/emails/collective.apply.for.host.hbs:
```diff
@@ -98,7 +98,7 @@ Dear admin of {{host.name}},
   <a
     class="btn blue"
-    href="{{config.host.website}}/{{host.slug}}/admin/pending-applications#application-{{collective.id}}"
+    href="{{config.host.website}}/{{host.slug}}/admin/host-applications?accountSlug={{collective.slug}}"
```
Not a fan of using the `slug` as collective admins can change it and the link would become invalid while the collective still exists and still has an application pending.

znarf replied:
Ok, let's send the id then.

gustavlrsn replied:
Good point, fixed this to use the id instead

Betree commented:
Would be great if we can have the frontend backward-compatible, to make sure links sent in previous emails still work

gustavlrsn commented:
> Would be great if we can have the frontend backward-compatible, to make sure links sent in previous emails still work

@Betree thanks for the review! Fixed the comments and updated the frontend PR to be fully backwards compatible.
